model/RepositoryService.php

```diff
@@ -219,4 +222,25 @@ protected function getNextVersion(string $resourceId): int
 
         return $candidate + 1;
     }
+
+    private function getUpdatedItemEventDispatcher(): UpdatedItemEventDispatcher
+    {
+        return $this->getServiceLocator()->get(UpdatedItemEventDispatcher::class);
+    }
+
+    private function getQtiService(): Service
+    {
+        return $this->getServiceLocator()->get(Service::class);
+    }
+
+    /**
+     * @throws common_Exception
+     */
+    private function updateItemsRelations(array $originFilesystemMap, Resource $resource): void
+    {
+        if (key($originFilesystemMap) === CreatorItems::PROPERTY_ITEM_CONTENT_URI) {
```

Comment:
Not directly related to your task, but we still need to handle the relation restoring when restoring shared stimulus / media too.

Reply:
i agree,  i will take into consideration